src/wallet/rpcwallet.cpp

```diff
@@ -806,6 +806,82 @@ UniValue movecmd(const JSONRPCRequest& request)
     return true;
 }
 
+UniValue rescan(const JSONRPCRequest& request)
+{
+    if (!EnsureWalletIsAvailable(request.fHelp))
+        return NullUniValue;
+
+    const int nParams = request.params.size();
+
+    if (request.fHelp || nParams > 1 || fPruneMode)
+        throw runtime_error(
+            "rescan ( \"height\" )\n"
+            "\nRescan the wallet for transactions\n"
+            "\nCurrently works only on non-pruned nodes\n"
```

Comment:
Can we insert a message between these two lines saying:

`WARNING: this operation may take a long time!`? Because when you scan a couple million blocks, it kinda takes a long time.

Reply:
Good idea!